### PR TITLE
chore: 0.13 update analytics with new segment library

### PR DESCRIPTION
**Pull request by mkhq**
**What this PR does / why we need it**:

This uses the new [Segment Analytics Next](https://github.com/segmentio/analytics-next) library which has improved handling for pending events on shutdown and better TypeScript support. This also includes a fix for nock and node 18 by overriding node-fetch in the test setup, details in https://github.com/nock/nock/issues/2336.

Did several rounds of manual testing using the quickstart example to ensure we record events Run Command and Command Result events while in dev console. Also verified the sessionId uses the parent session.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by eysi09 — DISMISSED** [review]
This is awesome! Thank you so much for fixing this and updating Segment. 

Massive improvement.

File: core/src/analytics/analytics.ts [hunk 1]
```diff
@@ -272,15 +273,13 @@ export class AnalyticsHandler {
     cloudUser?: UserResult
     ciInfo: CiInfo
   }) {
-    const segmentClient = require("analytics-node")
-    this.segment = new segmentClient(API_KEY, { flushAt: 20, flushInterval: 300 })
+    this.segment = new Analytics({ writeKey: API_KEY, maxEventsInBatch: 1, flushInterval: 3000 })
```
Comment (vvagaytsev) [inline core/src/analytics/analytics.ts]: Are these changes intended?
1. increase `flushInterval` from 300 to 300
2. `maxEventsInBatch = 1`

**Review by vvagaytsev — APPROVED** [review]
Looks good, thank you!

File: core/src/analytics/analytics.ts (hunk 1 above)
Comment (mkhq) [reply core/src/analytics/analytics.ts]: I think this comment was on an older version? There are 5 events in batch and the flushInterval was increased to avoid too frequent checking. The important thing is that analytics are flushed on process close which has been improved with the new library.
